### PR TITLE
Alerting: Fix simplified query step

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes two  bugs in the simplified query editor:

1. fixes refs not being editable when creating a new alert from a panel
2. fixes the switch not being set when clicking `Deactivate` button (this is a regression)

**Why do we need this feature?**

Its fixing 2 bugs.

**Who is this feature for?**
All users.

**Which issue(s) does this PR fix?**:
 
related to this [escalation](https://github.com/grafana/support-escalations/issues/13650)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 1]
```diff
@@ -167,14 +167,20 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
...
   const { isAdvancedMode, simpleCondition, setSimpleCondition } = useAdvancedMode(
-    editorSettings,
+    simplifiedQueryInForm,
     isGrafanaAlertingType,
     isNewFromQueryParams,
     dataQueries,
     expressionQueries
   );
 
+  useEffect(() => {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:180]: this syncs  `editorSettings.simplifiedQueryEditor` in the form with `isAdvancedmode`

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 2]
```diff
@@ -688,7 +694,7 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
         confirmText="Deactivate"
         icon="exclamation-triangle"
         onConfirm={() => {
-          setValue('editorSettings.simplifiedNotificationEditor', true);
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:691]: this fixes the bug not setting the switch to not advanced when confirming `deactivate`

Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:691]: I'm guessing the change from `simplifiedNotificationEditor` to `simplifiedQueryEditor` is what fixes the issue and is intentional, right?

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 3]
```diff
@@ -167,14 +167,20 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
...
-    editorSettings,
+    simplifiedQueryInForm,
     isGrafanaAlertingType,
     isNewFromQueryParams,
     dataQueries,
     expressionQueries
   );
 
+  useEffect(() => {
+    setValue('editorSettings.simplifiedQueryEditor', !isAdvancedMode);
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:181]: I think we should populate correct value in the `formValuesFromQueryParams` function instead

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx (hunk 2 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:691]: correct

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx (hunk 1 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx]: I've just pushed a commit with this simplification

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx (hunk 3 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx]: with the last commit, we don't need this useEffect anymore

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts [hunk 4]
```diff
@@ -28,15 +26,15 @@ function initializeSimpleCondition(
...
+  simplifiedQueryEditor: boolean | undefined,
   isGrafanaAlertingType: boolean,
   isNewFromQueryParams: boolean,
   dataQueries: Array<AlertQuery<ExpressionQuery | AlertDataQuery>>,
   expressionQueries: Array<AlertQuery<ExpressionQuery>>
 ) {
   const queryParamsAreTransformable = areQueriesTransformableToSimpleCondition(dataQueries, expressionQueries);
   return (
-    Boolean(editorSettings?.simplifiedQueryEditor) === false ||
+    Boolean(simplifiedQueryEditor) === false ||
```
[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts:40]: truee

[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts]: done!

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts [hunk 5]
```diff
@@ -66,10 +66,13 @@ export const useAdvancedMode = (
   );
 
   useEffect(() => {
-    if (!isAdvancedMode && isGrafanaAlertingType) {
-      setSimpleCondition(getSimpleConditionFromExpressions(expressionQueries));
+    if (isGrafanaAlertingType) {
+      setValue('editorSettings.simplifiedQueryEditor', !isAdvancedMode);
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts:70]: This code is being called every time we edit anything in expression queries. It shouldn't be a big issue but I'm a bit concerned about potential side effects of that. 
I guess we want to change this value only if the user actually switches the toggle

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts [hunk 6]
```diff
@@ -47,14 +46,15 @@ export function determineAdvancedMode(
...
    */
 export const useAdvancedMode = (
-  editorSettings: SimplifiedEditor | undefined,
+  simplifiedQueryEditor: boolean | undefined,
   isGrafanaAlertingType: boolean,
   isNewFromQueryParams: boolean,
   dataQueries: Array<AlertQuery<ExpressionQuery | AlertDataQuery>>,
   expressionQueries: Array<AlertQuery<ExpressionQuery>>
 ) => {
+  const { setValue } = useFormContext<RuleFormValues>();
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts:55]: I see some inconsistency here. We pass `simplifiedQueryEditor` as a parameter (which is effectively `editorSettings.simplifiedQueryEditor`) but at the same time we add a formContext as a dependency
If we want to keep the form context as a dependency, we don't need to pass most of the params. On the other hand I'd try not to have form dependencies in hooks

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/useAdvancedMode.ts]: this code has been simplified

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**